### ISSUE TITLE
[JENKINS-12875] "No valid crumb was included in the request" errors all around

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-15#1147]
<p>After upgrading Jenkins from 1.448 to 1.451 (with yum, upgraded plugins too) several parts of Jenkins have become unusable with error "No valid crumb was included in the request":</p>

<h4><a name="Consoleoutput"></a>Console output</h4>

<ul>
	<li>When building, the log is unavailable, see <span class="nobr"><a href="https://issues.jenkins.io/secure/attachment/21553/21553_console_log.png" title="console_log.png attached to JENKINS-12875">console_log.png<sup><img class="rendericon" src="https://issues.jenkins.io/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span></li>
	<li>Once the build is done, the complete log is available, but the same error appears on the side <span class="nobr"><a href="https://issues.jenkins.io/secure/attachment/21552/21552_console_side.png" title="console_side.png attached to JENKINS-12875">console_side.png<sup><img class="rendericon" src="https://issues.jenkins.io/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span></li>
</ul>

<h4><a name="Buildconfiguration"></a>Build configuration</h4>

<ul>
	<li>Builds cannot be launched anymore from the build page (The firebug error console displays the "No valid crumb was included in the request" error). They still can be launched from the dashboard though.</li>
	<li>Cannot add build steps (when choosing a build step type an error is reported in background), see <span class="nobr"><a href="https://issues.jenkins.io/secure/attachment/21556/21556_add_build_step.png" title="add_build_step.png attached to JENKINS-12875">add_build_step.png<sup><img class="rendericon" src="https://issues.jenkins.io/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span></li>
</ul>

<h4><a name="ManageJenkinsconfiguration"></a>Manage Jenkins configuration</h4>

<ul>
	<li>The page loads only partially (first half is permanently greyed out, see <span class="nobr"><a href="https://issues.jenkins.io/secure/attachment/21555/21555_configuration.png" title="configuration.png attached to JENKINS-12875">configuration.png<sup><img class="rendericon" src="https://issues.jenkins.io/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span>). The firebug error console displays the "No valid crumb was included in the request" error.</li>
</ul>

<h4><a name="ManageJenkinsplugins"></a>Manage Jenkins plugins</h4>

<ul>
	<li>Cannot upgrade nor downgrade, see error shown <span class="nobr"><a href="https://issues.jenkins.io/secure/attachment/21554/21554_plugins.png" title="plugins.png attached to JENKINS-12875">plugins.png<sup><img class="rendericon" src="https://issues.jenkins.io/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span></li>
</ul>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=naloy">naloy</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-12875" target="_blank">"No valid crumb was included in the request" errors all around</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Critical
<li><b>component(s)</b>: core
<li><b>label(s)</b>: 2.0, 2.0-planned, crumb
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2016-04-01T23:57:26+00:00
<li><b>votes</b>: 20
<li><b>watchers</b>: 43
<li><b>imported</b>: 2025-11-16
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>After upgrading Jenkins from 1.448 to 1.451 (with yum, upgraded plugins too) several parts of Jenkins have become unusable with error "No valid crumb was included in the request":</p>

<h4><a name="Consoleoutput"></a>Console output</h4>

<ul>
	<li>When building, the log is unavailable, see <span class="nobr"><a href="https://issues.jenkins.io/secure/attachment/21553/21553_console_log.png" title="console_log.png attached to JENKINS-12875">console_log.png<sup><img class="rendericon" src="https://issues.jenkins.io/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span></li>
	<li>Once the build is done, the complete log is available, but the same error appears on the side <span class="nobr"><a href="https://issues.jenkins.io/secure/attachment/21552/21552_console_side.png" title="console_side.png attached to JENKINS-12875">console_side.png<sup><img class="rendericon" src="https://issues.jenkins.io/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span></li>
</ul>

<h4><a name="Buildconfiguration"></a>Build configuration</h4>

<ul>
	<li>Builds cannot be launched anymore from the build page (The firebug error console displays the "No valid crumb was included in the request" error). They still can be launched from the dashboard though.</li>
	<li>Cannot add build steps (when choosing a build step type an error is reported in background), see <span class="nobr"><a href="https://issues.jenkins.io/secure/attachment/21556/21556_add_build_step.png" title="add_build_step.png attached to JENKINS-12875">add_build_step.png<sup><img class="rendericon" src="https://issues.jenkins.io/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span></li>
</ul>

<h4><a name="ManageJenkinsconfiguration"></a>Manage Jenkins configuration</h4>

<ul>
	<li>The page loads only partially (first half is permanently greyed out, see <span class="nobr"><a href="https://issues.jenkins.io/secure/attachment/21555/21555_configuration.png" title="configuration.png attached to JENKINS-12875">configuration.png<sup><img class="rendericon" src="https://issues.jenkins.io/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span>). The firebug error console displays the "No valid crumb was included in the request" error.</li>
</ul>

<h4><a name="ManageJenkinsplugins"></a>Manage Jenkins plugins</h4>

<ul>
	<li>Cannot upgrade nor downgrade, see error shown <span class="nobr"><a href="https://issues.jenkins.io/secure/attachment/21554/21554_plugins.png" title="plugins.png attached to JENKINS-12875">plugins.png<sup><img class="rendericon" src="https://issues.jenkins.io/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span></li>
</ul>
</pre>
</details>
</details>
<ul><li><i>environment</i>: <code>CentOS release 5.5 (Final) 64b</code></li></ul>
<details><summary><i>6 attachments</i></summary>

- [Screen Shot 2014-03-12 at 10.43.58 AM.png](https://issues.jenkins.io/secure/attachment/25542/Screen%20Shot%202014-03-12%20at%2010.43.58%20AM.png)
  > ![Screen Shot 2014-03-12 at 10.43.58 AM.png](https://issues.jenkins.io/secure/attachment/25542/Screen%20Shot%202014-03-12%20at%2010.43.58%20AM.png)
- [add_build_step.png](https://issues.jenkins.io/secure/attachment/21556/add_build_step.png)
  > ![add_build_step.png](https://issues.jenkins.io/secure/attachment/21556/add_build_step.png)
- [configuration.png](https://issues.jenkins.io/secure/attachment/21555/configuration.png)
  > ![configuration.png](https://issues.jenkins.io/secure/attachment/21555/configuration.png)
- [console_log.png](https://issues.jenkins.io/secure/attachment/21553/console_log.png)
  > ![console_log.png](https://issues.jenkins.io/secure/attachment/21553/console_log.png)
- [console_side.png](https://issues.jenkins.io/secure/attachment/21552/console_side.png)
  > ![console_side.png](https://issues.jenkins.io/secure/attachment/21552/console_side.png)
- [plugins.png](https://issues.jenkins.io/secure/attachment/21554/plugins.png)
  > ![plugins.png](https://issues.jenkins.io/secure/attachment/21554/plugins.png)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>onemanbucket's <a href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=169307&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-169307">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>I reproduced the problem with a simple nginx reverse proxy and csrf enabled.</p>

<p>The problem is that jenkins stores its' csrf token in a http header called '.crumb', AFAIK headers must only contain alphanumerics and dashes, and nginx will remove invalid headers from the request (unless configured not to).</p>

<p>Sooo, a workaround is to use ignore_invalid_headers in nginx or disable csrf protection until patched.</p>

<p>I'll post a pull request shortly</p>
</pre>
</details>

</details>
<p>I reproduced the problem with a simple nginx reverse proxy and csrf enabled.</p>

<p>The problem is that jenkins stores its' csrf token in a http header called '.crumb', AFAIK headers must only contain alphanumerics and dashes, and nginx will remove invalid headers from the request (unless configured not to).</p>

<p>Sooo, a workaround is to use ignore_invalid_headers in nginx or disable csrf protection until patched.</p>

<p>I'll post a pull request shortly</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>onemanbucket's <a href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=169310&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-169310">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Pull request: <a href="https://github.com/jenkinsci/jenkins/pull/596" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/pull/596</a></p>
</pre>
</details>

</details>
<p>Pull request: <a href="https://github.com/jenkinsci/jenkins/pull/596" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/pull/596</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ozawa.jpg" title="ozawa's avatar" /> ozawa's <a href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=179142&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-179142">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Today, I've had also the same issue. Delete jobs or clone from github hasn't worked correctly.
This problem is caused after upgrading 1.515.</p>
</pre>
</details>

</details>
<p>Today, I've had also the same issue. Delete jobs or clone from github hasn't worked correctly.<br/>
This problem is caused after upgrading 1.515.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/johnnygoods.jpg" title="johnnygoods's avatar" /> johnnygoods's <a href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=179717&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-179717">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Also seeing this issue when using jenkins 1.517 behind nginx reverse proxy. Setting nginx ignore_invalid_headers had no effect. Disabling CSRF did.</p>
</pre>
</details>

</details>
<p>Also seeing this issue when using jenkins 1.517 behind nginx reverse proxy. Setting nginx ignore_invalid_headers had no effect. Disabling CSRF did.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>vadakr's <a href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=196587&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-196587">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>To disable CSRF (<a href="https://issues.jenkins-ci.org/secure/attachment/25542/Screen%20Shot%202014-03-12%20at%2010.43.58%20AM.png" class="external-link" target="_blank" rel="nofollow noopener">Screenshot</a>) - </p>

<p>Jenkins &gt; Manage Jenkins &gt; Configure Global Security</p>

<p>At the very bottom, you will see a checkbox for "Prevent Cross Site Request Forgery exploits". Uncheck this and apply.</p>
</pre>
</details>

</details>
<p>To disable CSRF (<a href="https://issues.jenkins-ci.org/secure/attachment/25542/Screen%20Shot%202014-03-12%20at%2010.43.58%20AM.png" class="external-link" target="_blank" rel="nofollow noopener">Screenshot</a>) - </p>

<p>Jenkins > Manage Jenkins > Configure Global Security</p>

<p>At the very bottom, you will see a checkbox for "Prevent Cross Site Request Forgery exploits". Uncheck this and apply.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/daemoen.png" title="daemoen's avatar" /> daemoen's <a href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=199483&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-199483">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Tested multiple scenarios.</p>

<p>Following configs work:
OS: Centos 6 x86_64
Jenkins: 1.561 from Jenkins Repo</p>

<p>NGINX Reverse Proxy Setup test #1 (works):
<b><em>/etc/nginx/conf.d/jenkins.conf</em></b></p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">#
# Jenkins over SSL
#

upstream jenkins {
  server 127.0.0.1:8080 fail_timeout=0;
}

server {
  listen 80;
  server_name jenkins.domain.com;
  <span class="code-keyword">return</span> 301 https:<span class="code-comment">//$host$request_uri;
</span>}

server {
  ignore_invalid_headers off;
  listen 443;
  server_name jenkins.domain.com;

  ssl on;
  ssl_certificate /etc/nginx/ssl/wildcard.domain.chained.crt;
  ssl_certificate_key /etc/nginx/ssl/wildcard.domain.key;

  location / {
    proxy_set_header        Host $host;
    proxy_set_header        X-Real-IP $remote_addr;
    proxy_set_header        X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_redirect http:<span class="code-comment">// https://;
</span>    proxy_pass              http:<span class="code-comment">//jenkins;
</span>  }
}
</pre>
</div></div>
<p>NGINX Reverse Proxy Setup test #2 (does NOT work):
<b><em>/etc/nginx/conf.d/jenkins.conf</em></b></p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">#
# Jenkins over SSL
#

upstream jenkins {
  server 127.0.0.1:8080 fail_timeout=0;
}

server {
  listen 80;
  server_name jenkins.domain.com;
  <span class="code-keyword">return</span> 301 https:<span class="code-comment">//$host$request_uri;
</span>}

server {
  ignore_invalid_headers on; # Or simply leave it unspecified; This is the DEFAULT behavior of ignore_invalid_headers
  listen 443;
  server_name jenkins.domain.com;

  ssl on;
  ssl_certificate /etc/nginx/ssl/wildcard.domain.chained.crt;
  ssl_certificate_key /etc/nginx/ssl/wildcard.domain.key;

  location / {
    proxy_set_header        Host $host;
    proxy_set_header        X-Real-IP $remote_addr;
    proxy_set_header        X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_redirect http:<span class="code-comment">// https://;
</span>    proxy_pass              http:<span class="code-comment">//jenkins;
</span>  }
}
</pre>
</div></div>

<p>NGINX Reverse Proxy Setup test #3 (works):
<b><em>/etc/nginx/conf.d/jenkins.conf</em></b></p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">#
# Jenkins over SSL
#

upstream jenkins {
  server 127.0.0.1:8080 fail_timeout=0;
}

server {
  listen 80;
  server_name jenkins.domain.com;
  <span class="code-keyword">return</span> 301 https:<span class="code-comment">//$host$request_uri;
</span>}

server {
  ignore_invalid_headers on;
  listen 443;
  server_name jenkins.domain.com;

  ssl on;
  ssl_certificate /etc/nginx/ssl/wildcard.domain.chained.crt;
  ssl_certificate_key /etc/nginx/ssl/wildcard.domain.key;

  location / {
    proxy_set_header        Host $host;
    proxy_set_header        X-Real-IP $remote_addr;
    proxy_set_header        X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_redirect http:<span class="code-comment">// https://;
</span>    proxy_pass              http:<span class="code-comment">//jenkins;
</span>  }
}
</pre>
</div></div>
<p><b><em>/etc/sysconfig/jenkins</em></b></p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">JENKINS_JAVA_OPTIONS=<span class="code-quote">"-Djava.awt.headless=<span class="code-keyword">true</span> -Dhudson.security.csrf.requestfield=crumb"</span>
</pre>
</div></div>
</pre>
</details>

</details>
<p>Tested multiple scenarios.</p>

<p>Following configs work:<br/>
OS: Centos 6 x86_64<br/>
Jenkins: 1.561 from Jenkins Repo</p>

<p>NGINX Reverse Proxy Setup test #1 (works):<br/>
<b><em>/etc/nginx/conf.d/jenkins.conf</em></b></p>

<pre>
#
# Jenkins over SSL
#

upstream jenkins {
  server 127.0.0.1:8080 fail_timeout=0;
}

server {
  listen 80;
  server_name jenkins.domain.com;
  <span class="code-keyword">return</span> 301 https:<span class="code-comment">//$host$request_uri;
</span>}

server {
  ignore_invalid_headers off;
  listen 443;
  server_name jenkins.domain.com;

  ssl on;
  ssl_certificate /etc/nginx/ssl/wildcard.domain.chained.crt;
  ssl_certificate_key /etc/nginx/ssl/wildcard.domain.key;

  location / {
    proxy_set_headerHost $host;
    proxy_set_headerX-Real-IP $remote_addr;
    proxy_set_headerX-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_redirect http:<span class="code-comment">// https://;
</span>    proxy_pass      http:<span class="code-comment">//jenkins;
</span>  }
}
</pre>
<p>NGINX Reverse Proxy Setup test #2 (does NOT work):<br/>
<b><em>/etc/nginx/conf.d/jenkins.conf</em></b></p>

<pre>
#
# Jenkins over SSL
#

upstream jenkins {
  server 127.0.0.1:8080 fail_timeout=0;
}

server {
  listen 80;
  server_name jenkins.domain.com;
  <span class="code-keyword">return</span> 301 https:<span class="code-comment">//$host$request_uri;
</span>}

server {
  ignore_invalid_headers on; # Or simply leave it unspecified; This is the DEFAULT behavior of ignore_invalid_headers
  listen 443;
  server_name jenkins.domain.com;

  ssl on;
  ssl_certificate /etc/nginx/ssl/wildcard.domain.chained.crt;
  ssl_certificate_key /etc/nginx/ssl/wildcard.domain.key;

  location / {
    proxy_set_headerHost $host;
    proxy_set_headerX-Real-IP $remote_addr;
    proxy_set_headerX-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_redirect http:<span class="code-comment">// https://;
</span>    proxy_pass      http:<span class="code-comment">//jenkins;
</span>  }
}
</pre>

<p>NGINX Reverse Proxy Setup test #3 (works):<br/>
<b><em>/etc/nginx/conf.d/jenkins.conf</em></b></p>

<pre>
#
# Jenkins over SSL
#

upstream jenkins {
  server 127.0.0.1:8080 fail_timeout=0;
}

server {
  listen 80;
  server_name jenkins.domain.com;
  <span class="code-keyword">return</span> 301 https:<span class="code-comment">//$host$request_uri;
</span>}

server {
  ignore_invalid_headers on;
  listen 443;
  server_name jenkins.domain.com;

  ssl on;
  ssl_certificate /etc/nginx/ssl/wildcard.domain.chained.crt;
  ssl_certificate_key /etc/nginx/ssl/wildcard.domain.key;

  location / {
    proxy_set_headerHost $host;
    proxy_set_headerX-Real-IP $remote_addr;
    proxy_set_headerX-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_redirect http:<span class="code-comment">// https://;
</span>    proxy_pass      http:<span class="code-comment">//jenkins;
</span>  }
}
</pre>
<p><b><em>/etc/sysconfig/jenkins</em></b></p>

<pre>
JENKINS_JAVA_OPTIONS=<span class="code-quote">"-Djava.awt.headless=<span class="code-keyword">true</span> -Dhudson.security.csrf.requestfield=crumb"</span>
</pre>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/daemoen.png" title="daemoen's avatar" /> daemoen's <a href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=199546&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-199546">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Further testing of this shows that even with the working setups above, chrome still does not like it.  Firefox works fine.</p>
</pre>
</details>

</details>
<p>Further testing of this shows that even with the working setups above, chrome still does not like it.  Firefox works fine.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>elephantjim's <a href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=224288&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-224288">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>I'm using Jenkins behind a reverse proxy using nginx, too.  I tried to set ignore_invalid_headers to "off" on my server's nginx settings to no avail.  But then I found this: <a href="http://trac.nginx.org/nginx/ticket/117" class="external-link" target="_blank" rel="nofollow noopener">http://trac.nginx.org/nginx/ticket/117</a></p>

<p>It turns out that nginx needs to process headers before selecting a virtual server: the "Host" header determines which server to use.  Unless the ignore_invalid_headers is set to off on the default server, it won't get read.  The other way is to set it in nginx.conf in the http context.  That worked for me.</p>
</pre>
</details>

</details>
<p>I'm using Jenkins behind a reverse proxy using nginx, too.  I tried to set ignore_invalid_headers to "off" on my server's nginx settings to no avail.  But then I found this: <a href="http://trac.nginx.org/nginx/ticket/117" class="external-link" target="_blank" rel="nofollow noopener">http://trac.nginx.org/nginx/ticket/117</a></p>

<p>It turns out that nginx needs to process headers before selecting a virtual server: the "Host" header determines which server to use.  Unless the ignore_invalid_headers is set to off on the default server, it won't get read.  The other way is to set it in nginx.conf in the http context.  That worked for me.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jglick.png" title="jglick's avatar" /> jglick's <a href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=252138&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-252138">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Assigning to <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck" class="user-hover" rel="danielbeck">danielbeck</a> on the basis that he filed <a href="https://github.com/jenkinsci/jenkins/pull/1319" class="external-link" target="_blank" rel="nofollow noopener">PR 1319</a> which possibly just needs to have some merge conflicts resolved and be retested, but feel free to reassign if you think someone else should pick it up.</p>
</pre>
</details>

</details>
<p>Assigning to <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck" class="user-hover" rel="danielbeck">danielbeck</a> on the basis that he filed <a href="https://github.com/jenkinsci/jenkins/pull/1319" class="external-link" target="_blank" rel="nofollow noopener">PR 1319</a> which possibly just needs to have some merge conflicts resolved and be retested, but feel free to reassign if you think someone else should pick it up.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link's <a href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=252720&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-252720">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Code changed in jenkins
User: Daniel Beck
Path:
 core/src/main/java/hudson/security/csrf/CrumbFilter.java
 core/src/main/java/hudson/security/csrf/CrumbIssuer.java
 core/src/main/java/hudson/security/csrf/CrumbIssuerDescriptor.java
 core/src/main/java/hudson/security/csrf/DefaultCrumbIssuer.java
 test/src/test/groovy/hudson/model/SlaveTest.groovy
<a href="http://jenkins-ci.org/commit/jenkins/ed0ea631cd4af0e678da96d7297910ba1612749d" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/ed0ea631cd4af0e678da96d7297910ba1612749d</a>
Log:
  <span class="error">&#91;FIXED JENKINS-12875&#93;</span> Change default crumb name to Jenkins-Crumb</p>


</pre>
</details>

</details>
<p>Code changed in jenkins<br/>
User: Daniel Beck<br/>
Path:<br/>
 core/src/main/java/hudson/security/csrf/CrumbFilter.java<br/>
 core/src/main/java/hudson/security/csrf/CrumbIssuer.java<br/>
 core/src/main/java/hudson/security/csrf/CrumbIssuerDescriptor.java<br/>
 core/src/main/java/hudson/security/csrf/DefaultCrumbIssuer.java<br/>
 test/src/test/groovy/hudson/model/SlaveTest.groovy<br/>
<a href="http://jenkins-ci.org/commit/jenkins/ed0ea631cd4af0e678da96d7297910ba1612749d" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/ed0ea631cd4af0e678da96d7297910ba1612749d</a><br/>
Log:<br/>
  <span class="error">&#91;FIXED JENKINS-12875&#93;</span> Change default crumb name to Jenkins-Crumb</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link's <a href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=252721&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-252721">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Code changed in jenkins
User: Daniel Beck
Path:
 test/src/test/java/lib/form/RowSetTest.java
<a href="http://jenkins-ci.org/commit/jenkins/5b1642252ebd977c21832eb12cae633b3616429f" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/5b1642252ebd977c21832eb12cae633b3616429f</a>
Log:
  <a href="https://issues.jenkins.io/browse/JENKINS-12875" title="&quot;No valid crumb was included in the request&quot; errors all around" class="issue-link" data-issue-key="JENKINS-12875"><del>JENKINS-12875</del></a> Fix test</p>


</pre>
</details>

</details>
<p>Code changed in jenkins<br/>
User: Daniel Beck<br/>
Path:<br/>
 test/src/test/java/lib/form/RowSetTest.java<br/>
<a href="http://jenkins-ci.org/commit/jenkins/5b1642252ebd977c21832eb12cae633b3616429f" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/5b1642252ebd977c21832eb12cae633b3616429f</a><br/>
Log:<br/>
  <a href="https://issues.jenkins.io/browse/JENKINS-12875" title=""No valid crumb was included in the request" errors all around" class="issue-link" data-issue-key="JENKINS-12875"><del>JENKINS-12875</del></a> Fix test</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link's <a href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=252723&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-252723">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Code changed in jenkins
User: Daniel Beck
Path:
 core/src/main/java/hudson/security/csrf/CrumbIssuer.java
<a href="http://jenkins-ci.org/commit/jenkins/8b6d38de67154fb96c59791082d5cbe743b2b693" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/8b6d38de67154fb96c59791082d5cbe743b2b693</a>
Log:
  <a href="https://issues.jenkins.io/browse/JENKINS-12875" title="&quot;No valid crumb was included in the request&quot; errors all around" class="issue-link" data-issue-key="JENKINS-12875"><del>JENKINS-12875</del></a> Restrict access to constant</p>


</pre>
</details>

</details>
<p>Code changed in jenkins<br/>
User: Daniel Beck<br/>
Path:<br/>
 core/src/main/java/hudson/security/csrf/CrumbIssuer.java<br/>
<a href="http://jenkins-ci.org/commit/jenkins/8b6d38de67154fb96c59791082d5cbe743b2b693" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/8b6d38de67154fb96c59791082d5cbe743b2b693</a><br/>
Log:<br/>
  <a href="https://issues.jenkins.io/browse/JENKINS-12875" title=""No valid crumb was included in the request" errors all around" class="issue-link" data-issue-key="JENKINS-12875"><del>JENKINS-12875</del></a> Restrict access to constant</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link's <a href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=252724&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-252724">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Code changed in jenkins
User: Daniel Beck
Path:
 core/src/main/java/hudson/security/csrf/CrumbFilter.java
 core/src/main/java/hudson/security/csrf/CrumbIssuer.java
 core/src/main/java/hudson/security/csrf/CrumbIssuerDescriptor.java
 core/src/main/java/hudson/security/csrf/DefaultCrumbIssuer.java
 test/src/test/groovy/hudson/model/SlaveTest.groovy
 test/src/test/java/lib/form/RowSetTest.java
<a href="http://jenkins-ci.org/commit/jenkins/a87c38c5052cba745127a0205c40490b30c8bdbf" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/a87c38c5052cba745127a0205c40490b30c8bdbf</a>
Log:
  Merge pull request #2198 from daniel-beck/<a href="https://issues.jenkins.io/browse/JENKINS-12875" title="&quot;No valid crumb was included in the request&quot; errors all around" class="issue-link" data-issue-key="JENKINS-12875"><del>JENKINS-12875</del></a></p>

<p><span class="error">&#91;FIX JENKINS-12875&#93;</span> Change default crumb name to Jenkins-Crumb</p>


<p>Compare: <a href="https://github.com/jenkinsci/jenkins/compare/4649e04cbe58...a87c38c5052c" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/compare/4649e04cbe58...a87c38c5052c</a></p>

</pre>
</details>

</details>
<p>Code changed in jenkins<br/>
User: Daniel Beck<br/>
Path:<br/>
 core/src/main/java/hudson/security/csrf/CrumbFilter.java<br/>
 core/src/main/java/hudson/security/csrf/CrumbIssuer.java<br/>
 core/src/main/java/hudson/security/csrf/CrumbIssuerDescriptor.java<br/>
 core/src/main/java/hudson/security/csrf/DefaultCrumbIssuer.java<br/>
 test/src/test/groovy/hudson/model/SlaveTest.groovy<br/>
 test/src/test/java/lib/form/RowSetTest.java<br/>
<a href="http://jenkins-ci.org/commit/jenkins/a87c38c5052cba745127a0205c40490b30c8bdbf" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/a87c38c5052cba745127a0205c40490b30c8bdbf</a><br/>
Log:<br/>
  Merge pull request #2198 from daniel-beck/<a href="https://issues.jenkins.io/browse/JENKINS-12875" title=""No valid crumb was included in the request" errors all around" class="issue-link" data-issue-key="JENKINS-12875"><del>JENKINS-12875</del></a></p>

<p><span class="error">&#91;FIX JENKINS-12875&#93;</span> Change default crumb name to Jenkins-Crumb</p>


<p>Compare: <a href="https://github.com/jenkinsci/jenkins/compare/4649e04cbe58...a87c38c5052c" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/compare/4649e04cbe58...a87c38c5052c</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jknurek.jpg" title="jknurek's avatar" /> jknurek's <a href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=254373&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-254373">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Maybe it's worth noting in this ticket:
I installed Jenkins 2.0-rc-1 and created a view using Build Monitor View plugin (1.8+build.201601112328), and get this error (which leaves the UI in an infinite "sorry" loop</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">Apr 14, 2016 8:16:03 AM hudson.security.csrf.CrumbFilter doFilter
WARNING: No valid crumb was included in request <span class="code-keyword">for</span> /$stapler/bound/18964be5-e86d-4242-a57b-35ee573d52c9/fetchJobViews. Returning 403.
</pre>
</div></div>
</pre>
</details>

</details>
<p>Maybe it's worth noting in this ticket:<br/>
I installed Jenkins 2.0-rc-1 and created a view using Build Monitor View plugin (1.8+build.201601112328), and get this error (which leaves the UI in an infinite "sorry" loop</p>


<pre>
Apr 14, 2016 8:16:03 AM hudson.security.csrf.CrumbFilter doFilter
WARNING: No valid crumb was included in request <span class="code-keyword">for</span> /$stapler/bound/18964be5-e86d-4242-a57b-35ee573d52c9/fetchJobViews. Returning 403.
</pre>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=254374&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-254374">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jknurek" class="user-hover" rel="jknurek">jknurek</a> That's unrelated to this issue and looks like a bug in the plugin that cannot handle the CSRF protection option (around since 2010 or so) being enabled.</p>
</pre>
</details>

</details>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jknurek" class="user-hover" rel="jknurek">jknurek</a> That's unrelated to this issue and looks like a bug in the plugin that cannot handle the CSRF protection option (around since 2010 or so) being enabled.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jknurek.jpg" title="jknurek's avatar" /> jknurek's <a href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=254597&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-254597">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>My first thought was that it was a bug specific to the Build Monitor View plugin, but when I run Jenkins 1.651.1, the same configuration works ok. I'm only getting this error with the 2.0-rc-1. 
(but I'm not sure what the underlying issue is in this case, so it is likely a separate issue)</p>
</pre>
</details>

</details>
<p>My first thought was that it was a bug specific to the Build Monitor View plugin, but when I run Jenkins 1.651.1, the same configuration works ok. I'm only getting this error with the 2.0-rc-1. <br/>
(but I'm not sure what the underlying issue is in this case, so it is likely a separate issue)</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/janek.png" title="janek's avatar" /> janek's <a href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=255957&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-255957">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck" class="user-hover" rel="danielbeck">danielbeck</a> Build Monitor has supported CSRF protection since its early days <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/> </p>

<p>The problem is caused by change <a href="https://github.com/jenkinsci/jenkins/compare/4649e04cbe58...a87c38c5052c" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/compare/4649e04cbe58...a87c38c5052c</a> , which replaced the hard-coded '.crumb' header with 'Jenkins-Crumb', that's why they no longer match.</p>

<p>I can work around that in Build Monitor.</p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jknurek" class="user-hover" rel="jknurek">jknurek</a> you can watch the progress of this issue on github at <a href="https://github.com/jan-molak/jenkins-build-monitor-plugin/issues/215" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jan-molak/jenkins-build-monitor-plugin/issues/215</a>, which is the official issue tracker for Build Monitor.</p>

<p>Thanks!
Jan</p>
</pre>
</details>

</details>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck" class="user-hover" rel="danielbeck">danielbeck</a> Build Monitor has supported CSRF protection since its early days <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/> </p>

<p>The problem is caused by change <a href="https://github.com/jenkinsci/jenkins/compare/4649e04cbe58...a87c38c5052c" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/compare/4649e04cbe58...a87c38c5052c</a> , which replaced the hard-coded '.crumb' header with 'Jenkins-Crumb', that's why they no longer match.</p>

<p>I can work around that in Build Monitor.</p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jknurek" class="user-hover" rel="jknurek">jknurek</a> you can watch the progress of this issue on github at <a href="https://github.com/jan-molak/jenkins-build-monitor-plugin/issues/215" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jan-molak/jenkins-build-monitor-plugin/issues/215</a>, which is the official issue tracker for Build Monitor.</p>

<p>Thanks!<br/>
Jan</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=255963&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-255963">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=janek" class="user-hover" rel="janek">janek</a></p>

<blockquote><p>which replaced the hard-coded '.crumb' header with 'Jenkins-Crumb', that's why they no longer match.</p></blockquote>

<p><a href="https://wiki.jenkins-ci.org/display/JENKINS/Features+controlled+by+system+properties" class="external-link" target="_blank" rel="nofollow noopener">The crumb name has been configurable since 1.310</a> (<tt>hudson.security.csrf.requestfield</tt>), and that was necessary for anyone running Jenkins behind nginx, as it dropped the <tt>.crumb</tt> header (<a href="https://issues.jenkins.io/browse/JENKINS-12875" title="&quot;No valid crumb was included in the request&quot; errors all around" class="issue-link" data-issue-key="JENKINS-12875"><del>JENKINS-12875</del></a>). So we renamed the default to something nginx accepts. Maybe it's a good idea to check which crumb name is used by Jenkins in your plugin to adapt to user configuration.</p>
</pre>
</details>

</details>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=janek" class="user-hover" rel="janek">janek</a></p>

<blockquote><p>which replaced the hard-coded '.crumb' header with 'Jenkins-Crumb', that's why they no longer match.</p></blockquote>

<p><a href="https://wiki.jenkins-ci.org/display/JENKINS/Features+controlled+by+system+properties" class="external-link" target="_blank" rel="nofollow noopener">The crumb name has been configurable since 1.310</a> (<tt>hudson.security.csrf.requestfield</tt>), and that was necessary for anyone running Jenkins behind nginx, as it dropped the <tt>.crumb</tt> header (<a href="https://issues.jenkins.io/browse/JENKINS-12875" title=""No valid crumb was included in the request" errors all around" class="issue-link" data-issue-key="JENKINS-12875"><del>JENKINS-12875</del></a>). So we renamed the default to something nginx accepts. Maybe it's a good idea to check which crumb name is used by Jenkins in your plugin to adapt to user configuration.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>nvn's <a href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=261967&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261967">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Hi Jan, yeah I think the issue is either with Jenkins or may be with Tfs-jenkins plugin, I am not sure. Pls suggest.</p>

<p>Thanks</p>
</pre>
</details>

</details>
<p>Hi Jan, yeah I think the issue is either with Jenkins or may be with Tfs-jenkins plugin, I am not sure. Pls suggest.</p>

<p>Thanks</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>12niner's <a href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=273684&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-273684">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Having just wasted a few hours debugging why I was seeing this: if you re-install Jenkins you need to clear your cookies or you'll hit this.</p>
</pre>
</details>

</details>
<p>Having just wasted a few hours debugging why I was seeing this: if you re-install Jenkins you need to clear your cookies or you'll hit this.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-12875?focusedId=273734&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-273734">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=12niner" class="user-hover" rel="12niner">12niner</a> Too many cookies is tracked as <a href="https://issues.jenkins.io/browse/JENKINS-25046" title="Cookie header too long, causing a 413 HTTP error" class="issue-link" data-issue-key="JENKINS-25046">JENKINS-25046</a>.</p>
</pre>
</details>

</details>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=12niner" class="user-hover" rel="12niner">12niner</a> Too many cookies is tracked as <a href="https://issues.jenkins.io/browse/JENKINS-25046" title="Cookie header too long, causing a 413 HTTP error" class="issue-link" data-issue-key="JENKINS-25046">JENKINS-25046</a>.</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Originally duplicated by: <a href="https://github.com/timja-org/jenkins-gh-issues-poc-11-15/issues?q=in%3Atitle%20JENKINS-25116">JENKINS-25116</a>]</i>

**Comment by jenkins-infra-bot** [comment]
<i>[Originally duplicated by: <a href="https://github.com/timja-org/jenkins-gh-issues-poc-11-15/issues?q=in%3Atitle%20JENKINS-23793">JENKINS-23793</a>]</i>

**Comment by jenkins-infra-bot** [comment]
<i>[Originally related to: <a href="https://github.com/timja-org/jenkins-gh-issues-poc-11-15/issues?q=in%3Atitle%20JENKINS-7518">JENKINS-7518</a>]</i>

**Comment by jenkins-infra-bot** [comment]
<i>[Originally related to: <a href="https://github.com/timja-org/jenkins-gh-issues-poc-11-15/issues?q=in%3Atitle%20JENKINS-12787">JENKINS-12787</a>]</i>
